src/KafkaNET.Library/Consumers/PartitionTopicInfo.cs

```diff
@@ -215,9 +212,47 @@ public long FetchOffset
             }
         }
 
+        /// <summary>
+        /// It happens when it gets consumer offset out of range exception. Try to fix the offset.
+        /// </summary>        
+        internal void ResetOffset(long resetOffset)
+        {
+            lock(this.fetchedOffsetLock)
```

Comment:
Do we need the lock here? ResetOffset() for a particular pti would be called only from one point (i.e. its FetcherRunnable)